soc/arm/st_stm32/stm32wba/power.c

```diff
@@ -100,16 +160,38 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	switch (state) {
 	case PM_STATE_SUSPEND_TO_IDLE:
 		if (substate_id <= 2) {
+			/* Erratum 2.2.15:
+			 * Enable ICACHE when exiting stop mode
+			 */
+			LL_ICACHE_SetMode(LL_ICACHE_1WAY);
+			LL_ICACHE_Enable();
+			while (LL_ICACHE_IsEnabled() == 0U) {
+			}
+
 			LL_LPM_DisableSleepOnExit();
 			LL_LPM_EnableSleep();
 		} else {
 			LOG_DBG("Unsupported power substate-id %u",
 							substate_id);
 		}
-	case PM_STATE_STANDBY:
-		/* To be tested */
-		LL_LPM_EnableSleep();
+		break;
 	case PM_STATE_SUSPEND_TO_RAM:
+#if defined(CONFIG_PM_S2RAM)
+		stm32wba_init();
+
+		LL_LPM_DisableSleepOnExit();
+		LL_LPM_EnableSleep();
+
+		/* Enabling  Ultra Low power mode */
+		LL_PWR_EnableUltraLowPowerMode();
+
+		LL_FLASH_EnableSleepPowerDown();
```

Comment:
Should you call `stm32_power_init()` directly as well ?

Reply:
Done